twistlock/tests/test_twistlock.py

```diff
@@ -97,11 +89,18 @@ def test_config_project(aggregator, fixture_group):
         aggregator.assert_metric_has_tag(metric, project_tag)
 
 
-def test_err_response(aggregator):
+def test_err_response(aggregator, instance):
 
     check = TwistlockCheck('twistlock', {}, [instance])
 
     with pytest.raises(Exception, match='^Error in response'):
         with mock.patch('requests.get', return_value=MockResponse('{"err": "invalid credentials"}'), autospec=True):
 
             check.check(instance)
+
+
+@pytest.mark.e2e
+def test_e2e(dd_agent_check, aggregator, instance):
+    with pytest.raises(Exception):
+        dd_agent_check(instance)
+    aggregator.assert_service_check("twistlock.license_ok", AgentCheck.CRITICAL)
```

Comment:
Why not check `twistlock.can_connect`?

Reply:
I tried it at first. But the I think this line here makes the check stop executing before it gets there:
https://github.com/DataDog/integrations-core/blob/master/twistlock/datadog_checks/twistlock/twistlock.py#L84

So it checks for the license and then stops. At least that's what I think

Reply:
I think that's fine then. Looks like it would require mocking data in order to get past `license = self._retrieve_json("/api/v1/settings/license")` on [line 78](https://github.com/DataDog/integrations-core/blob/master/twistlock/datadog_checks/twistlock/twistlock.py#L78), so `twistlock.license_ok` should be ok.

Reply:
I think one thing to make the test a bit more robust is checking that the `warning` is logged. An example can be found here: https://github.com/DataDog/integrations-core/blob/master/iis/tests/test_iis.py#L53

Reply:
I wasn't able to get caplog working with `dd_agent_check`. Caplog would always be empty. I made a new assertion instead that will assert that the exception is thrown because it failed at the expected endpoint. In this case `/api/v1/settings/license` endpoint.